lib/Module/KModule.cpp

```diff
@@ -390,9 +390,17 @@ void KModule::prepare(const Interpreter::ModuleOptions &opts,
   case eSwitchTypeLLVM:  pm3.add(createLowerSwitchPass()); break;
   default: klee_error("invalid --switch-type");
   }
+  RequiredOperandsAreScalarCheckPass *roascp =
+      new RequiredOperandsAreScalarCheckPass();
   pm3.add(new IntrinsicCleanerPass(*targetData));
   pm3.add(new PhiCleanerPass());
+  pm3.add(roascp);
   pm3.run(*module);
+
+  // Enforce the operand type invariants that the Executor expects
+  if (!roascp->checkPassed()) {
+    klee_error("Unexpected vector operands detected");
```

Comment:
Perhaps we ought to rename the pass then because the pass name implies it only checks that some operands are of scalar type. Suggestions for a name? `InstructionOperandTypeCheckPass`?